sdk/provisioning/Azure.Provisioning.CloudMachine/src/Client/CloudMachine.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+
+namespace Azure.CloudMachine;
+
+public partial class CloudMachineClient
+{
+    public string Id { get; }
+
+    public TokenCredential Credential { get; } = new ChainedTokenCredential(
+        new AzureDeveloperCliCredential()
+    );
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0007:DO provide a minimal constructor that takes only the parameters required to connect to the service.", Justification = "<Pending>")]
+    public CloudMachineClient(DefaultAzureCredential? credential = default, IConfiguration? configuration = default)
+    {
+        if (credential != default)
+        {
+            Credential = credential;
+        }
+
+        string? cmid;
+        if (configuration == default)
+        {
+            cmid = Azd.ReadOrCreateCmid();
+        }
+        else
+        {
+            cmid = configuration["CloudMachine:ID"];
+            if (cmid == null)
+            throw new Exception("CloudMachine:ID configuration value missing");
+        }
+
+        Id = cmid!;
+    }
+
+    protected CloudMachineClient()
+    {
+        Id = "CM";
+    }
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public ClientCache ClientCache { get; } = new ClientCache();
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public CloudMachineProperties Properties => new CloudMachineProperties(this);
+
+    public struct CloudMachineProperties
+    {
+        private readonly CloudMachineClient _cm;
+
+        internal CloudMachineProperties(CloudMachineClient cm) => _cm = cm;
+        public Uri DefaultContainerUri => new Uri($"https://{_cm.Id}.blob.core.windows.net/default");
+        public Uri BlobServiceUri => new Uri($"https://{_cm.Id}.blob.core.windows.net/");
```

Comment:
This feels a little risky long term.  Making sure you have one globally unique name can be tricky enough, but I don't know how folks would rationalize getting the unique name across multiple Azure services.  Eventually you should make it the infrastructure's problem to give you back the URIs and it'll be easier to run this in other clouds.

Reply:
Yeah, I need to think about it, but there is something nice about uniform names.